pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -555,14 +532,7 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 		readWrite = "READ_ONLY"
 	}
 
-	attachOp, err := gce.manager.AttachDiskOnCloudProvider(
-		disk, readWrite, instance.Zone, instance.Name)
-
-	if err != nil {
-		return mc.Observe(err)
-	}
-
-	return gce.manager.WaitForZoneOp(attachOp, instance.Zone, mc)
+	return mc.Observe(gce.manager.AttachDiskOnCloudProvider(disk, readWrite, instance.Zone, instance.Name))
```

Comment:
We're "Observing" metrics no matter what now instead of only on error. Wont this emit a ton of "nil"'s? Ditto everywhere else

Reply:
The old logic observes the operation regardless of whether an error exists, and this is done inside `WaitForZoneOp()`.

Reply:
Note that metrics capture latency as well as error/success.